src/goto-symex/symex_assign.cpp

```diff
@@ -485,12 +486,12 @@ void goto_symext::symex_assign_symbol(
               << messaget::eom;
     });
 
-  // Temporarily add the state guard
-  guard.emplace_back(state.guard.as_expr());
+  const exprt assignment_guard =
+    make_and(state.guard.as_expr(), conjunction(guard));
```

Comment:
Presumably this was done this way in the name of performance. Have you checked that your change doesn't noticeably slow symex down?

Reply:
Before this was adding an exprt to a vector then using this vector to make the `and_exprt` of the conjunction, now we first make the conjunction and then add the exprt to the vector of the `and_exprt` operands. So the operation are the same, just in a different order, this shouldn't impact performances.

Reply:
@owen-jones-diffblue I just noticed that my change could lead to `assignment_guard` being `TRUE && TRUE` so I've added a commit to have `make_and` handle the constant cases, can you check that?

Reply:
It seems good, once I saw from `exprt::is_true()` that boolean constant `exprt`s must be either `true_exprt` or `false_exprt`.